python-sdk/nuscenes/eval/predict/tests/test_metrics.py

```diff
@@ -0,0 +1,227 @@
+                                                    [10.63014, 10.64014, 9.21954]]), atol=1e-3, rtol=1e-3)
+
+    def test_stack_ground_truth(self):
+        value = metrics.stack_ground_truth(np.ones((5, 2)), 10)
+        np.testing.assert_equal(value, np.ones((10, 5, 2)))
+
+    def test_desired_number_of_modes_one_mode(self):
+        results = np.ones((10, 1))
+        value = metrics.desired_number_of_modes(results, [1, 5, 15, 25])
+        np.testing.assert_equal(value, np.ones((10, 4)))
+
+    def test_desired_number_of_modes_enough_data(self):
+        results = np.arange(75).reshape(3, 25)
+        value = metrics.desired_number_of_modes(results, [1, 5, 15, 25])
+        np.testing.assert_equal(value, np.array([[0, 4, 14, 24],
+                                                 [25, 29, 39, 49],
+                                                 [50, 54, 64, 74]]))
+
+    def test_desired_number_of_modes_not_enough(self):
+        results = np.arange(30).reshape(2, 15)
+        value = metrics.desired_number_of_modes(results, [1, 5, 15, 25])
+        np.testing.assert_equal(value, np.array([[0, 4, 14, 14],
+                                                 [15, 19, 29, 29]]))
+
+class TestAggregators(unittest.TestCase):
+
+    def test_RowMean(self):
+        rm = metrics.RowMean()
+        value = rm(np.arange(20).reshape(2, 10))
+        self.assertListEqual(value, [5, 6, 7, 8, 9, 10, 11, 12, 13, 14])
+
+        self.assertDictEqual(rm.serialize(), {'name': 'RowMean'})
+
+class TestMetrics(unittest.TestCase):
+
+    def test_MinADEK(self):
+        min_ade = metrics.MinADEK([1, 5, 10], [metrics.RowMean()])
+        self.assertDictEqual(min_ade.serialize(), {'name': 'MinADEK',
+                                                   'k_to_report': [1, 5, 10],
+                                                    'aggregators': [{'name': 'RowMean'}]})
+
+    def test_MinFDEK(self):
+        min_fde = metrics.MinFDEK([1, 5, 10], [metrics.RowMean()])
+        self.assertDictEqual(min_fde.serialize(), {'name': 'MinFDEK',
+                                                   'k_to_report': [1, 5, 10],
+                                                    'aggregators': [{'name': 'RowMean'}]})
+
+    def test_HitRateTopK(self):
+        hit_rate = metrics.HitRateTopK([1, 5, 10], [metrics.RowMean()], 2)
+        self.assertDictEqual(hit_rate.serialize(), {'k_to_report': [1, 5, 10],
+                                                    'name': 'HitRateTopK',
+                                                    'aggregators': [{'name': 'RowMean'}],
+                                                    'tolerance': 2})
+
+    def test_DeserializeMetric(self):
+
+        config = {'name': 'MinADEK',
+                  'k_to_report': [1, 5, 10],
+                  'aggregators': [{'name': 'RowMean'}]}
+
+        m = metrics.DeserializeMetric(config)
+        self.assertEqual(m.name, 'MinADEK')
+        self.assertListEqual(m.k_to_report, [1, 5, 10])
```

Comment:
The Metric class does not have a member called `k_to_report`.

Reply:
Changed the return type of DeserializeMetric